File: Day15-Tree Transformation.cpp
#include<bits/stdc++.h>
using namespace std;
//Problem link 
//https://practice.geeksforgeeks.org/problems/fbcd1787378ed396a8f24b47872cbc0ad2624f1d/1

class Solution {
    public:
    int solve(int n, vector<int>p){
        // code here
        int i,ans=n-1;
        vector<int>degree(n+1,0);
        for(i=1;i<n;i++){
            degree[p[i]]++;
            degree[i]++;
        }
        for(i=0;i<n;i++)if(degree[i]==1)ans--;
        
        if(ans<0)return 0;
        return ans;
    }
};

File: Day15-Tree Transformation.java
//Problem link 
//https://practice.geeksforgeeks.org/problems/fbcd1787378ed396a8f24b47872cbc0ad2624f1d/1
 import java.util.*;

class Solution {
    public static int solve(int N, int[] p) {
        int[] arr = new int[N];
        Arrays.fill(arr, 0);
        
        for (int i = 1; i < N; i++) {
            arr[p[i]]++;
            arr[i]++;
        }
        
        int count = 0;
        for (int i = 0; i < N; i++) {
            if (arr[i] == 1) {
                count++;
            }
        }
        
        if (count == N) {
            return 0;
        }
        
        return N - 1 - count;
    }
}